### ISSUE TITLE
Invalid spell targets shouldn't provide green highlight

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#2470]
At the moment, when you're casting a spell, most areas of the page light up with a green border when you mouse over them, regardless of the spell you're casting. It's not until you try to e.g. cast Pickpocket on one of your fellow party members that you get the "invalid target" error.

A green box highlight is a pretty clear "yes, go for it!" cue, so we should change this. Possibilities (in all cases, no green box):
- Don't do anything when you mouse over an invalid target. Cursor stays arrow with plus, waiting for you to find a valid target.
- Change the cursor to the "no" circle like you get when trying to add a task without a title.
- Shade over the target in a transparent red or gray field, clearly showing "not valid" (even to colorblind users).

_edit by Alys, June 2016:_

This should also apply to challenge tasks, which can no longer have skills cast on them, although at the moment, that's not obvious (you click on a skill then on a challenge task, but you then see the results of the skill being reversed and you get an error message). It might be nice if, when you hover over a challenge task for more than the time it takes to move the mouse past it, then you see a message explaining that you can't cast skills on challenge tasks.
##

**Comment by deilann** [comment]
Be careful with the color grey you choose, if that's how you go. Some greys may be the same as some of the blues.

Frex: http://s.wsj.net/public/resources/images/OB-VG023_health_DV_20121105173546.jpg (at least, I think that's the right one. it's hard to pull them off google images because there are also blank ones)

**Comment by deilann** [comment]
If you're looking for a colorblind-safe color, yellow is the best. Anyone who can see color at all can see yellow.

**Comment by SabreCat** [comment]
Maybe a yellow-black striping, like a no-parking zone? xD

**Comment by deilann** [comment]
That's a good idea!

**Comment by SabreCat** [comment]
I'm not sophisticated enough in CSS yet to know how I'd do that, unfortunately. I could research... or maybe a style guru like @brightbold or @nafoster could venture a suggestion?

**Comment by deilann** [comment]
I know I saw someone in the Tavern struggling with local install who stated
they wanted to do CSS stylesheets... maybe they'd like to take a crack at
this.

**Comment by SabreCat** [comment]
I _think_ that was @brightbold, but I might be thinking of a different Tavern conversation than you.

**Comment by deilann** [comment]
No, it was another user. They just posted in the dev guild 13 hours ago,
asking for help getting MongoDB running.

**Comment by SabreCat** [comment]
Ah, excellent.

**Comment by brightbold** [comment]
The guy Ryan is thinking of is Griffin Wiebel. He posted in the developers guild asking for help with Grunt and MongoDB. I don't think I'm likely to be able to help him but I responded just so someone would have. If @SabreCat or @colegleason or someone who actually knows what they're doing could help him out, maybe we could get another  CSSer on board.

**Comment by SabreCat** [comment]
Cool. I'll drop by there later, see if I can lend a hand.

**Comment by SabreCat** [comment]
Ah, looks like you already directed him to Aspiring Legends. I'll give it a bit and see if he drops in there, I don't normally check the Developers Guild.

**Comment by lefnire** [comment]
I vote:

> Don't do anything when you mouse over an invalid target. Cursor stays arrow with plus, waiting for you to find a valid target.

Then we don't have to futz with css

**Comment by Caspid** [comment]
To decrease the amount of trial-and-error mouseovering, the opacity of non-eligible items could decrease and all eligible items could be given the border.

Perhaps unrelated, but it's unclear whether skills are meant to apply to tasks or to ratings (+ / -).  When a spell is selected, removing the ratings would make the selection process more straightforward and less confusing.

**Comment by ghost** [comment]
In a separate but related issue, when you cast a skill and then mouseover a task (Such as Brutal Smash on a task), the task is highlighted at the border OR the empty checkbox turns into a check mark, depending on where you mouseover. It should only be highlighted, and should not cause a checkmark to appear.

Edit: The above behavior is on a browser. On my android phone using habitica as a webapp, the same thing happens.

**Comment by librarianmage** [comment]
Any progress on this?

**Comment by Alys** [comment]
Just the discussion you see here. The issue still exists.

**Comment by librarianmage** [comment]
Ok!

**Comment by lemoness** [comment]
I agree that the behavior should be that only valid targets are highlighted, not anything else. Shall we add the "Help welcome now" tag?

**Comment by paglias** [comment]
@lemoness go ahead!
Il 07 giu 2016 11:33 PM, "lemoness" notifications@github.com ha scritto:

**Comment by Alys** [comment]
I'm putting this on hold because we're most of the way through a project to change the website's front-end code and this might no longer happen afterwards, or if it does, there'll be new code that will need to be modified to fix it.

**Comment by lemoness** [comment]
This is handled differently now with the new site, and I don't believe that it is a major issue anymore, so I am closing this ticket. A new one can be opened again if necessary.
